docs: v0.3 restructure (Diátaxis) with v0.2 redirects (#115)
platform-mesh/platform-mesh.github.io: concepts/api-sharing.md

- Permission claims are part of the provider-consumer trust boundary and should be accepted intentionally.

## Permission claims

A provider often needs bounded access to a few resources outside its own API — to write a `Secret` with connection credentials, read a credential the consumer placed in the workspace, or emit `events`. Permission claims are the consent mechanic for that access: the provider declares what it needs on the APIExport; the consumer accepts each claim explicitly on the APIBinding. Neither side can unilaterally escalate.

Claims can be scoped by verb (read-only versus full access) and, in principle, by object selector. Platform Mesh itself is bootstrapped through this mechanism — the `core.platform-mesh.io` APIExport in `platform-mesh-system` claims access to workspaces, apibindings, logical clusters, and secrets so the account-operator can reconcile them on every account's behalf.

For the full Platform Mesh examples, see [API sharing reference](/reference/components/kcp/api-sharing.md). For field-level semantics, see [permission claims](https://docs.kcp.io/kcp/main/concepts/apis/exporting-apis/#permission-claims) in the kcp docs.

## Virtual workspace

> A provider with many consumer bindings cannot watch each consumer workspace separately. kcp solves this by publishing a *virtual workspace* endpoint per APIExport — a single wildcard view that aggregates all bound objects across consumers. Provider controllers connect to that one endpoint, see every relevant object annotated with its source workspace, and write status back through the same path.

Reviewer: ```suggestion
## Virtual Workspaces

A provider with many consumer bindings cannot watch each consumer workspace separately. kcp solves this by publishing *virtual workspace* endpoints — wildcard views that aggregates all bound objects across consumers on one kcp shard. Provider controllers connect to that one endpoint, see every relevant object annotated with its source workspace, and write status back through the same path.
```

Since there isn't a single completely aggregated view over all shards - aggregating from these VWs is what multicluster-runtime accomplishes.

> Almost every Platform Mesh operator (account-operator, security-operator, rebac-authz-webhook, the GraphQL gateway, marketplace, and per-service operators) consumes a virtual workspace this way. Controllers don't construct URLs by hand; they read them from `APIExportEndpointSlice.status`.

Reviewer: ```suggestion
Every Platform Mesh operator (account-operator, security-operator, rebac-authz-webhook, the GraphQL gateway, marketplace, and per-service operators) consumes virtual workspaces this way. Controllers don't construct URLs by hand; they read them from `APIExportEndpointSlice.status`.
```

## How Platform Mesh layers on this

The kcp primitives are generic. Platform Mesh layers account structure and lifecycle wiring on top:

- **Where APIExports live.** Platform Mesh provisions dedicated provider workspaces under a known path. The platform owner controls those locations; providers do not pick arbitrary paths.
- **Who consumes.** Consumer workspaces are mapped to [Accounts](./account-model.md) in the Platform Mesh hierarchy. APIBindings live inside a consumer Account's workspace.
- **Authorization wiring.** When a binding is activated, Platform Mesh updates the consumer Account's [IAM store](./identity-and-authorization.md) so the new API surfaces are covered by OpenFGA alongside the rest of the workspace.

A bare APIExport without Platform Mesh wiring works fine in vanilla kcp but will not appear in the marketplace, will not participate in IAM enforcement, and will not be discoverable through the Platform Mesh Portal.

## Upstream kcp ownership

kcp owns APIExport, APIBinding, APIResourceSchema, permission claim, identity, and virtual workspace semantics.

> Use upstream kcp documentation for canonical behavior:

Reviewer: ```suggestion
Reference upstream kcp documentation for canonical behavior:
```
